File: 0974. Subarray Sums Divisible by K.cpp
class Solution
{
public:
    int subarraysDivByK(vector<int>& nums, int k)
    {
        int ans = 0;
        int mod = 0;

        vector<int> v(k);
        v[0] = 1;

        for (int num : nums)
        {
            mod = (mod + num % k + k) % k;
            ans += v[mod];
            v[mod]++;
        }

        return ans;
    }
};

File: 0974. Subarray Sums Divisible by K.cs
public class Solution
{
    public int SubarraysDivByK(int[] nums, int k)
    {
        var modFrequencies = new Dictionary<int, int>
        {
            { 0, 1 }
        };
        var prefixMod = 0;
        var result = 0;

        foreach (var num in nums)
        {
            prefixMod += num % k + k;
            prefixMod %= k;

            if (modFrequencies.TryGetValue(prefixMod, out var count))
            {
                result += count;
                ++modFrequencies[prefixMod];
            }
            else
            {
                modFrequencies.Add(prefixMod, 1);
            }
        }

        return result;
    }
}